src/Crdt.Core/ChangeEntity.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Crdt.Core;
+
+public class ChangeEntity<TChange>
+{
+    [JsonConstructor]
+    public ChangeEntity()
+    {
+    }
+
+    public required int Index { get; set; }
```

Comment:
this is used to order changes within a single commit so that the ordering is consistent with how they were created when we retrieve changes out of the DB